ecs/basicsystem.go

```diff
@@ -0,0 +1,79 @@
+package ecs
+
+import "runtime"
+
+// BasicSystemWrapper is the default implementation of the System interface.
+type BasicSystemWrapper struct {
```

Comment:
Well, if it's in a struct, then there's no way to 'override' those methods. This problem doesn't exist for interfaces.